Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -1259,6 +1259,27 @@ void NetworkStorageManager::syncLocalStorage(CompletionHandler<void()>&& complet
     });
 }
 
+void NetworkStorageManager::fetchLocalStorage(CompletionHandler<void(HashMap<WebCore::ClientOrigin, HashMap<String, String>>&&)>&& completionHandler)
+{
+    ASSERT(RunLoop::isMain());
+    ASSERT(!m_closed);
+
+    protectedWorkQueue()->dispatch([this, protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler)]() mutable {
+        assertIsCurrent(workQueue());
+
+        HashMap<WebCore::ClientOrigin, HashMap<String, String>> localStorageMap;
+
+        for (auto& [clientOrigin, manager] : m_originStorageManagers) {
+            if (auto* localStorageManager = manager->existingLocalStorageManager())
+                localStorageMap.add(clientOrigin, localStorageManager->fetchStorageMap());
+        }
+
+        RunLoop::protectedMain()->dispatch([completionHandler = WTFMove(completionHandler), localStorageMap = crossThreadCopy(WTFMove(localStorageMap))] mutable {
+            completionHandler(WTFMove(localStorageMap));
```

Comment:
Is the `crossThreadCopy()` needed here? I added it because we're moving the HashMap to a different thread--but if we're using `WTFMove()` and not using the HashMap on this work thread afterwards, do we still need the `crossThreadCopy()`?

Reply:
Yes; `crossThreadCopy` knows to avoid copy if possible, and it should be always invoked.